server/src/main/java/org/opensearch/search/query/ConcurrentQueryPhaseSearcher.java

```diff
@@ -77,19 +76,18 @@ private static boolean searchWithCollectorManager(
             collectorManager = QueryCollectorManagerContext.createQueryCollectorManager(collectorContexts);
         }
 
-        try {
-            final ReduceableSearchResult result = searcher.search(query, collectorManager);
-            result.reduce(queryResult);
-        } catch (EarlyTerminatingCollector.EarlyTerminationException e) {
-            queryResult.terminatedEarly(true);
-        } catch (TimeExceededException e) {
+        final ReduceableSearchResult result = searcher.search(query, collectorManager);
+        result.reduce(queryResult);
```

Comment:
We should still keep the `try-catch` block here and catch the `RuntimeException` and unwrap it along with `ExecutionException` to throw the internal exception to the caller. This will keep it consistent with the non concurrent path and avoid resulting in `NotSerializableExceptionWrapper` to the caller. Can you check if that is what is happening indeed when `RuntimeException` is thrown ?

Reply:
I believe the `TimeExceededException` won't be thrown here anymore, however the `TaskCancelledException` that is thrown by the `cancel_after_time_interval` case does see this behavior. We aren't catching `TaskCancelledException` in the non-concurrent case anyways though.

I think you are right that we do need to keep the catch for `EarlyTerminatingCollector.EarlyTerminationException` though (or rather catch the `RuntimeException` and rethrow).

Reply:
Agree with @sohami , moreover the non-concurrent case also has this exception handling logic